### ISSUE TITLE
bug: Bad placement of the buttons in the bottom-navigation in iOS PWA mode 

### ISSUE THREAD

**Issue by EzHeisenberg** [saadeghi/daisyui#1732]
### What version of daisyUI are you using?

v.2.51.5

### Describe your issue

While developing a web application on **iPhone** in PWA mode, I used the Bottom-navigation component, I realized that there was a bad placement of the navigation buttons container when using the tag for iOS : 

```html
The <meta name="viewport" content="viewport-fit=cover"> is required for responsive of the bottom navigation in iOS.
```

here are some examples, we can see here that icons, texts or both not a good position

![IMG_3550](https://user-images.githubusercontent.com/41981936/229696884-2f1b68d9-b9fd-4f0a-953c-29f5f8a91d62.jpg)
![IMG_3549](https://user-images.githubusercontent.com/41981936/229696890-b13b772a-94d9-40ac-ad91-b7a32dfdab93.jpg)
![IMG_3548](https://user-images.githubusercontent.com/41981936/229696894-b372d6c0-e3e8-4d43-a8e4-8d95f8f408d2.jpg)

### What browsers are you seeing the problem on?

Safari

**Comment by saadeghi** [comment]
Can you give me a code example to test?  
Also can you try increasing the height of `btm-nav` using a `h-*` tailwind utility class? I think your items are bigger than the available height.

**Comment by EzHeisenberg** [comment]
Bien sur, je l'ai utilisé dans un cas basique. mais dans les `h-*` 

### HTML
```html
<!DOCTYPE html>
<html lang="fr">
<head>
    <meta charset="UTF-8">
    <title>title</title>
    <link rel="icon" href="">
    <meta name="viewport" content="viewport-fit=cover, width=device-width, height=device-height, initial-scale=1"/>
</head>
    <body>
        <div class="container mx-auto max-w-5xl p-4">
            <div class="text-sm breadcrumbs pb-4 text-gray-500">
                <ul>
                    <li>
                        <a href="#">Accueil</a>
                    </li>
                </ul>
            </div>
        </div>
        <div class="btm-nav">
            <a href="#">
                <em class="fa-solid fa-house"></em>
                <span class="btm-nav-label">
                      Home
                  </span>
            </a>
        
            <a href="#" class="active text-primary">
                <em class="fa-solid fa-cubes"></em>
                <span class="btm-nav-label">
                      Projets
                  </span>
            </a>
        
            <a href="#">
                <em class="fa-solid fa-ticket"></em>
                <span class="btm-nav-label">
                      Frais
                  </span>
            </a>
        
            <a href="#">
                <em class="fa-solid fa-user"></em>
                <span class="btm-nav-label">
                      Profil
                  </span>
            </a>
        </div>
    </body>
</html>
```

### CSS

```css
body{
    min-height: 100vh;
    background-color: #f4f4f4;
}
```

**Comment by EzHeisenberg** [comment]
when I apply a `h-*` the height changes as expected, except that it doesn't look very nice because the elements are centered vertically and the bottom-navigation block looks big.


# h-28
![IMG_3551](https://user-images.githubusercontent.com/41981936/229703623-d3e422cf-b268-4932-a0fb-77245ad39194.PNG)

# h-24
![IMG_3552](https://user-images.githubusercontent.com/41981936/229703688-c6a071aa-c2a2-443d-8a2c-81210bfe96dd.PNG)



# h-20
![IMG_3553](https://user-images.githubusercontent.com/41981936/229703701-c54929ab-6ef5-4789-9fae-340a69dfc330.PNG)



---


On the other hand, on Chrome browsers, brave... the expected behavior of the navigation bar is the right one 

<img width="372" alt="image" src="https://user-images.githubusercontent.com/41981936/229702816-e6344730-6a57-43d2-8363-9a744b25e6c4.png">

**Comment by saadeghi** [comment]
`.btm-nav` has this style `padding-bottom:env(safe-area-inset-bottom);` to make space for iOS "home bar" (that line on the bottom). without that padding, your menu items would go under that line so I think you should keep that padding. the value of padding is `env(safe-area-inset-bottom)` but you can change it using utility classes like `pb-4` or `pb-2` or anything. let me know if that fixes the problem for you.

**Comment by thobson** [comment]
I'm seeing similar behaviour (using standard btm-nav) during scroll (when the home bar is not visible). Please see the screenshot. Do I need to somehow watch for window resize events and apply a new padding if the home bar is not visible? thanks

With home bar:

![IMG_7619](https://github.com/saadeghi/daisyui/assets/208345/90b1a20e-42d8-43fe-9ec7-d90f7aa6a500)

Without home bar:

![IMG_7620](https://github.com/saadeghi/daisyui/assets/208345/8de1f45e-a777-4392-95fe-5dc2161b5fa6)

**Comment by thobson** [comment]
Strangely, the issue seems to be with the use of `env(safe-area-inset-bottom)` at least in the latest iOS version. Using `pb-0` works for me.

**Comment by raphaelbadia** [comment]
Hello, a possible workaround is to add `h-auto` to the parent div and a bit of padding in the children if necessary

before: 
```html
<div className="btm-nav z-50 bg-base-100 drop-shadow-sm">
   .......
</div>
```
<img width="423" alt="before" src="https://github.com/saadeghi/daisyui/assets/1901628/1970fb70-8843-4d7a-b4bb-31df7d649f95">


after:
```html
<div className="btm-nav z-50 bg-base-100 drop-shadow-sm h-auto">
   .......
</div>
```
<img width="423" alt="after with h-auto" src="https://github.com/saadeghi/daisyui/assets/1901628/e54ff7fd-61b7-4e26-b7b4-e69ccf778932">

**Comment by dillonfagan** [comment]
I'm having the same issue with my app when it's installed as a PWA on iOS.
Using `h-auto` does not solve the issue for me. It only reduces the height of the navbar when the app is viewed in-browser.
Adding top padding causes the nav indicator to render on top of the nav items.

```html
<div class="btm-nav lg:hidden">
```

![Screenshot 2023-10-04 at 4 22 56 PM](https://github.com/saadeghi/daisyui/assets/5107025/c6a540df-6eac-4bd4-8fdd-2e6cf4d9649d)

**Comment by dillonfagan** [comment]
I think I found a solution: add `box-content` to the same container using the `btm-nav` style.

That will allow the height of the box to increase when the bottom padding is applied. The problem was that the height was fixed to 64px.

**Comment by lgraubner** [comment]
The `box-content` fix works fine for me. I could provide a PR if wanted. Let me know/assign me. :)

**Comment by MaazDev** [comment]
you can also see this bug on the daisyui website on ios https://daisyui.com/components/bottom-navigation/
a PR would be good

**Comment by pioz** [comment]
I've this problem when loading a pwa with the btm-nav. I can confirm that adding `box-content` will fix the problem. 

BEFORE
![IMG_5812](https://github.com/user-attachments/assets/f9131497-4e07-4c0a-b876-acb84873ed67)

AFTER
![IMG_5811](https://github.com/user-attachments/assets/fcefe998-67ba-432a-b178-653234950573)

**Comment by ndkv** [comment]
Yes, `box-content` does it, brilliant (and, in hindsight, logical)!

**Comment by saadeghi** [comment]
`btm-nav` in daisyUI 4 had a lot of issues. We replaced it in the new version (daisyUI 5 beta) with `dock` which fixes these kind of issues, has a better design and it's more customizable.
